rmw_dds_common/include/rmw_dds_common/context.hpp

```diff
@@ -49,6 +49,151 @@ struct Context
   rmw_guard_condition_t * listener_thread_gc;
   /// Guard condition that should be triggered when the graph changes.
   rmw_guard_condition_t * graph_guard_condition;
+
+  using publish_callback_t =
+    std::function<rmw_ret_t(const rmw_publisher_t * pub, const void * msg)>;
+  /// Publish a graph message when updating or destroying graph cache.
+  publish_callback_t publish_callback;
+
+  /// Update graph for creating a node.
+  /**
+   * \param name node name.
+   * \param namespace_ node namespace.
+   * \return `RMW_RET_OK` if successful, or
+   * \return `RMW_RET_ERROR` an unexpected error occurs.
+   */
+  RMW_DDS_COMMON_PUBLIC
+  rmw_ret_t
+  update_node_graph(
+    const std::string & name, const std::string & namespace_);
+
+  /// Update graph for destroying a node.
+  /**
+   * \param name node name.
+   * \param namespace_ node namespace.
+   * \return `RMW_RET_OK` if successful, or
+   * \return `RMW_RET_ERROR` an unexpected error occurs.
+   */
+  RMW_DDS_COMMON_PUBLIC
+  rmw_ret_t
+  destroy_node_graph(
```

Comment:
related to @eboasson 's comment https://github.com/ros2/rmw_dds_common/pull/73#pullrequestreview-1687016197, it sounds to me like destroy entire node graph object... maybe `add(update)_node_graph` and `remove_node_graphy` would be better? (this does not block the PR.)